middle_end/inlining_transforms.ml

```diff
@@ -71,6 +71,36 @@ let fold_over_projections_of_vars_bound_by_closure ~closure_id_being_applied
       function_decls)
     init
 
+let fold_over_projections_of_vars_bound_by_closures
+      ~closure_id_being_applied
+      ~lhs_of_application
+      ~(function_decls : Flambda.function_declarations)
+      ~init
+      ~f =
+  let function_names = Variable.Map.keys function_decls.funs in
+  Variable.Map.fold
+    (fun _function_var (function_decl : Flambda.function_declaration) acc ->
+       let (--) = Variable.Set.diff in
+       let function_free_vars = function_decl.free_variables in
+       let function_params = Parameter.Set.vars function_decl.params in
+       let bound_variables =
+         function_free_vars -- function_params -- function_names
```

Comment:
This should probably have parentheses to make the intent more obvious.

Reply:
Indeed.